ext/sockets/sockets.c

```diff
@@ -1871,6 +1872,11 @@ PHP_FUNCTION(socket_set_option)
 			const char l_onoff_key[] = "l_onoff";
 			const char l_linger_key[] = "l_linger";
 
+			if (Z_TYPE_P(arg4) != IS_ARRAY) {
+				zend_argument_type_error(4, "must be of type array, %s given", zend_zval_value_name(arg4));
+				RETURN_THROWS();
+			}
+	
 			convert_to_array(arg4);
```

Comment:
This appears to be superfluous, since we just checked that it is an array (and otherwise bailed out).